### PR TITLE
Start reporting isolate life cycle events

### PR DESCRIPTION
**Pull request by johnmccutchan**
- [x] Add IsolateTracker.
- [x] Tell the IsolateTracker what the view's isolate id is.
- [x] Start printing important isolate life cycle events.

### CODE REVIEW

**Comment by johnmccutchan** [comment]
@devoncarew

File: packages/flutter_tools/lib/src/isolate_tracker.dart [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+// Track and report isolate life cycle events.
+class IsolateTracker {
+  IsolateTracker(this.serviceProtocol);
+
+  // TODO(johnmccutchan): There can be more than one view.
+  String _viewIsolateId;
+  set viewIsolateId(String viewIsolateId) {
+    _viewIsolateId = viewIsolateId;
+    printStatus('View is now controlled by isolate $_viewIsolateId');
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/isolate_tracker.dart]: `printTrace`?

File: packages/flutter_tools/lib/src/isolate_tracker.dart [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    if (!kInterestingEventKinds.contains(event.kind))
+      return;
+    String messagePrefix = 'Isolate ${event.isolate.id}:';
+    if (_isViewIsolate(event.isolate.id)) {
+      messagePrefix = 'View isolate ${event.isolate.id}:';
+    }
+    if (event.kind == 'PauseExit') {
+      // TODO(johnmccutchan): Load the isolate and get any error.
+    }
+    printStatus('$messagePrefix ${event.kind}');
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/isolate_tracker.dart]: Probably printTrace here - status should be for user facing messages.

File: packages/flutter_tools/lib/src/isolate_tracker.dart [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      return;
+    }
+    _isolateEventSubscription =
+        serviceProtocol.onIsolateEvent.listen(onEvent);
+    _debugEventSubscription =
+        serviceProtocol.onDebugEvent.listen(onEvent);
+  }
+
+  Future<Null> stop() async {
+    _isolateEventSubscription?.cancel();
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/isolate_tracker.dart]: we should `await` these

File: packages/flutter_tools/lib/src/isolate_tracker.dart [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    if (_isViewIsolate(event.isolate.id)) {
+      messagePrefix = 'View isolate ${event.isolate.id}:';
+    }
+    if (event.kind == 'PauseExit') {
+      // TODO(johnmccutchan): Load the isolate and get any error.
+    }
+    printStatus('$messagePrefix ${event.kind}');
+  }
+
+  Future<Null> start() async {
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/isolate_tracker.dart]: Does this need to be async or return a Future?

**Comment by devoncarew** [comment]
lgtm, w/ some questions

**Comment by eseidelGoogle** [comment]
Looking through old PRs, found this one. Curious of the status?

**Comment by johnmccutchan** [comment]
@eseidelGoogle This is on hold until I lay some more foundation work

**Comment by Hixie** [comment]
@johnmccutchan Since you're not going to be working on this for a few days I'm going to close the PR for now so that it isn't in our review queue. Please don't hesitate to reopen it when you're ready for review.

**Pull request closed without merging**